d2/src/main/java/com/linkedin/d2/balancer/dualread/UriPropertiesDualReadMonitor.java

```diff
@@ -74,21 +81,29 @@ public void reportData(String clusterName, UriProperties property, boolean fromN
       }
     }
 
+    if (cluster._matched != cluster._uris)
+    {
+      RATE_LIMITED_LOGGER.info("Mismatched cluster properties for {} (match score {}):\nOld LB: {}\nNew LB: {}",
+          clusterName, cluster._matched / cluster._uris, cluster._oldLb, cluster._newLb);
+    }
```

Comment:
Could this log be at the end of the method? since cluster._uris still need to include the new lb's uri count at line 90, and also it makes more sense to read the log at 95 first (when debug level log enabled) then this message.

Reply:
I'm updating this if it sounds good to you.

Reply:
Makes sense